report/marshal.go

```diff
@@ -118,3 +123,86 @@ func MakeFromBytes(buf []byte) (*Report, error) {
 	}
 	return &rep, nil
 }
+
+// MakeFromFile construct a rReport from a file, with the encoding
+// determined by the extension (".msgpack" or ".json", with an
+// optional ".gz").
+func MakeFromFile(path string) (rpt Report, _ error) {
+	f, err := os.Open(path)
+	if err != nil {
+		return rpt, err
+	}
+	defer f.Close()
+
+	handle, gzipped, err := fileType(path)
+	if err != nil {
+		return rpt, err
+	}
+
+	var buf []byte
+	if gzipped {
+		r, err := gzip.NewReader(f)
+		if err != nil {
+			return rpt, err
+		}
+		buf, err = ioutil.ReadAll(r)
+	} else {
+		buf, err = ioutil.ReadAll(f)
+	}
+	if err != nil {
+		return rpt, err
+	}
+	err = rpt.ReadBytes(buf, handle)
+
+	return rpt, err
+}
+
+func (rep *Report) WriteToFile(path string, compressionLevel int) error {
```

Comment:
lint error "exported method Report.WriteToFile should have comment or be unexported"